buildsystem/Doxygen.cmake

```diff
@@ -0,0 +1,66 @@
+# Copyright 2014-2015 the openage authors. See copying.md for legal info.
+
+option(GENERATE_DOCS "Generate documentation from the source files." ON)
+
+set(DOXYFILE_TEMPLATE "${BUILDSYSTEM_DIR}/templates/Doxyfile.in")
+set(DOCS_ROOT_DIRECTORY "${CMAKE_BINARY_DIR}/docs")
+
+function(add_doxygen_target TARGET TARGET_SRC_DIR)
+	if(NOT DOXYGEN_FOUND)
+		return()
+	endif()
+
+	set(DOXYFILE "${CMAKE_BINARY_DIR}/${TARGET}_Doxyfile")
+	set(OUTPUT_DIRECTORY "${DOCS_ROOT_DIRECTORY}/${TARGET}")
```

Comment:
In my opinion, libopenage and openage should have different doc folders. But if you think otherwise, the only change required in this implementation would be to change https://github.com/jprashanth/openage/blob/buildsystem/CMakeLists.txt#L75-L76 to `add_doxygen_target("openage" libopenage/ openage/)`. This would create both the python and c++ docs together in the folder `bin/docs/openage`